src/blob_storage.cc

```diff
@@ -24,6 +24,26 @@ Status BlobStorage::NewPrefetcher(uint64_t file_number,
                                     result);
 }
 
+Status BlobStorage::DeleteBlobsInRanges(const RangePtr* ranges, size_t n,
+                                        bool include_end,
+                                        SequenceNumber obsolete_sequence) {
+  MutexLock l(&mutex_);
+  for (size_t i = 0; i < n; i++) {
+    auto begin = ranges[i].start, end = ranges[i].limit;
+    auto cmp = cf_options_.comparator;
+
+    for (auto it = blob_ranges_.lower_bound(*begin);
+         it != blob_ranges_.upper_bound(*end); it++) {
+      if (it->second->is_obsolete()) continue;
+      if ((include_end && cmp->Compare(it->second->largest_key(), *end) <= 0) ||
```

Comment:
assert `smallest_key()` is empty or `>= begin`

Reply:
why assert `smallest_key()` is empty? it should be included in `>= begin`.

Reply:
and `>= begin` is a promise of `lower_bound`, is it really necessary?

Reply:
My understanding is, `smallest_key()` is empty means it's of legacy manifest and so assume to be cover the full key range (more like `smallest_key() < begin` for any `begin`). Am I missing something?

Reply:
Just an assert to verify our data-structure is correct in test or debug mode.